hudi-common/src/main/java/org/apache/hudi/common/model/HoodieAvroRecordMerger.java

```diff
@@ -36,17 +36,28 @@ public class HoodieAvroRecordMerger implements HoodieRecordMerger {
 
   @Override
   public String getMergingStrategy() {
-    return StringUtils.DEFAULT_MERGER_STRATEGY_UUID;
+    return HoodieRecordMerger.DEFAULT_MERGER_STRATEGY_UUID;
   }
 
   @Override
-  public Option<HoodieRecord> merge(HoodieRecord older, HoodieRecord newer, Schema schema, Properties props) throws IOException {
+  public Pair<Option<HoodieRecord>, Schema> merge(HoodieRecord older, Schema oldSchema,
+      HoodieRecord newer, Schema newSchema, Properties props) throws IOException {
     ValidationUtils.checkArgument(older.getRecordType() == HoodieRecordType.AVRO);
     ValidationUtils.checkArgument(newer.getRecordType() == HoodieRecordType.AVRO);
     if (older instanceof HoodieAvroRecord && newer instanceof HoodieAvroRecord) {
-      return Option.of(preCombine(older, newer));
+      HoodieRecord res = preCombine(older, newer);
```

Comment:
@wzx140 i understand what you're doing here, but we shouldn't be guessing the use-case (either de-duping w/in a batch or merging w/ already persisted record) based on the type of the record we see.

 - It's extremely brittle 
 - It's very unintuitive

Instead we can pass special config (for ex, only when we do de-duping) and then control whether we do pre-combine or full-blown merge based on that